### PR TITLE
Macos support

### PR DESCRIPTION
**Pull request by dommiller88**
updated dependencies
added overall macOS support (box example works)

### CODE REVIEW

File: src/framework/mod.rs [hunk 1]
```diff
@@ -7,4 +7,8 @@ use crate::components::BoxComponent;
 
 pub fn run_app(tree: BoxComponent) {
     start_event_loop(tree);
-}
\ No newline at end of file
+}
+
+// pub fn run_app_mac(tree: BoxComponent) {
```
Comment (nicballesteros) [inline src/framework/mod.rs]: remove this commented out code

File: src/platform/macos.rs [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+            let command_buffer = command_queue.new_command_buffer();
+            command_buffer.present_drawable(drawable);
+            command_buffer.commit();
+        }
+        
+    });
+}
+
+/// Renders a rectangle that occupies exactly half of the canvas
+#[cfg(all(target_os = "macos", feature = "metal"))]
```
Comment (nicballesteros) [inline src/platform/macos.rs]: we should get rid of this draw function here. This is the job of the orchestrator.

File: src/platform/mod.rs [hunk 3]
```diff
@@ -4,6 +4,12 @@ pub mod windows;
 #[cfg(feature = "windows")]
 pub use windows::start_event_loop;
 
+#[cfg(feature = "macos")]
```
Comment (nicballesteros) [inline src/platform/mod.rs:7]: I don't think we'll need these two lines bc 
```
pub use windows::start_event_loop
```
exposes start_event_loop in the scope of platform. We only want to call platform::start_event_loop and not access platform::macos::some_func
Maybe we will need this in the future but for now, I don't think we need it.

File: src/platform/macos.rs [hunk 4]
```diff
@@ -0,0 +1,178 @@
+use crate::orchestrator::redraw::handle_redraw;
+use crate::orchestrator::event::click::handle_click;
+use crate::components::BoxComponent;
+
+// #[cfg(not(target_os = "macos"))]
+// fn main() {
+//     println!("This example is only supported on macos")
+// }
+
+// #[cfg(all(target_os = "macos", not(feature = "metal")))]
```
Comment (nicballesteros) [inline src/platform/macos.rs]: I think we should remove these lines.

File: examples/box/main.rs [hunk 5]
```diff
@@ -27,4 +32,20 @@ fn build() {
     );
 
     run_app(red_box);
-}
\ No newline at end of file
+}
+
+// fn build_mac() {
```
Comment (nicballesteros) [inline examples/box/main.rs]: remove these commented out lines

File: examples/box/main.rs [hunk 6]
```diff
@@ -7,11 +7,16 @@ use one1fy::framework::components::{
 
 // This function is only defined here because we are using windows.
 // Otherwise, Swift or Andoird NDK will call build() directly.
-#[cfg(feature = "windows")]
+// #[cfg(feature = "windows")]
```
Comment (nicballesteros) [inline examples/box/main.rs]: make this #[cfg(any(feature = "windows", feature = "macos"))]

File: Cargo.toml [hunk 7]
```diff
@@ -6,10 +6,17 @@ edition = "2021"
...
 [features]
-default = ["windows"]
 windows = ["skia-safe/gl", "skia-safe/d3d"]
+macos = ["skia-safe/gl", "skia-safe/metal"]
 
 [dependencies]
 skia-safe = "0.53.0"
 glutin = "0.29"
 gl = "0.14.0"
+cocoa = "0.24.0"
```
Comment (nicballesteros) [inline Cargo.toml:16]: double check we need all of these dependencies or if some of them are dev dependencies

File: src/platform/mod.rs (hunk 3 above)
Comment (dommiller88) [reply src/platform/mod.rs:7]: I tried removing these lines and the code did not run anymore

Comment (nicballesteros) [reply src/platform/mod.rs:7]: Interesting, we can leave it then

File: examples/box/main.rs [hunk 8]
```diff
@@ -7,11 +7,16 @@ use one1fy::framework::components::{
 
 // This function is only defined here because we are using windows.
 // Otherwise, Swift or Andoird NDK will call build() directly.
-#[cfg(feature = "windows")]
+#[cfg(any(feature = "windows", feature = "macos"))]
 fn main() {
     build();
 }
 
+// #[cfg(feature = "macos")]
```
Comment (nicballesteros) [inline examples/box/main.rs]: 💊 Now that we've fixed the duplicated code, we should delete this commented out code.

**Comment by nicballesteros** [comment]
You've still missed a few spots where theres commented out code.

**Review by nicballesteros — APPROVED** [review]

**Pull request merged**